### PR TITLE
Support observability for workers and error reporting of services

### LINKED ISSUE #17218: Improve OOTB observability to observe workers
**Description:**
Observability for jBallerina was introduced with #17079 
This feature should be extended to support worker interactions.

### LINKED ISSUE #17219: Service invocation errors should be reflected in OOTB observability
**Description:**

Observability for jBallerina was introduced with #17079
If a service panics during an invocation, the error details are not reflected in the server span or metrics

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
Support observability for workers
Attach logs to active spans
Support attaching of errors to service tracers and service metrics

Fixes #17219
Fixes #17218

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-backend-jvm-old/src/main/ballerina/compiler_backend_jvm/jvm_observability_gen.bal [hunk 1]
```diff
@@ -20,6 +20,13 @@ function emitStopObservationInvocation(jvm:MethodVisitor mv, int strandIndex) {
         io:sprintf("(L%s;)V", STRAND), false);
 }
 
+function emitReportErrorInvocation(jvm:MethodVisitor mv, int strandIndex, int errorIndex) {
+    mv.visitVarInsn(ALOAD, strandIndex);
+    mv.visitVarInsn(ALOAD, errorIndex);
+    mv.visitMethodInsn(INVOKESTATIC, "org/ballerinalang/jvm/observability/ObserveUtils", "reportError",
```
Comment (vinok88) [inline compiler/ballerina-backend-jvm-old/src/main/ballerina/compiler_backend_jvm/jvm_observability_gen.bal]: Better to move these class names to constants. Let's do it in a separate PR.

Comment (Anoukh) [reply compiler/ballerina-backend-jvm-old/src/main/ballerina/compiler_backend_jvm/jvm_observability_gen.bal]: Will send another PR
